File: src/main/scala/com/example/igor/viewholders/ImageWithoutTextViewHolder.scala
package com.example.igor.viewholders

import android.view.View
import android.widget.ImageView
import com.example.igor.amazingview.AmazingViewHolder
import rx.functions.Action1

class ImageWithoutTextViewHolder(itemView: View) extends AmazingViewHolder(itemView) with Action1[Integer] {

  import com.example.igor.R

  val image: ImageView = itemView.findViewById(R.id.custom_image).asInstanceOf[ImageView]

  def setImageResource(resId: Int): ImageWithoutTextViewHolder = {
    image.setImageResource(resId)
    this
  }

  def call(resId: Integer) {
    image.setImageResource(resId)
  }

  protected def onClickCallback: String = {
    null
  }
}

File: src/main/java/com/example/ImageWithoutTextViewHolder.java
package com.example;

import android.view.View;
import android.widget.ImageView;
import butterknife.Bind;
import butterknife.ButterKnife;
import com.example.amazingview.AmazingViewHolder;
import com.example.igor.R;
import rx.functions.Action1;

public class ImageWithoutTextViewHolder extends AmazingViewHolder implements Action1<Integer> {

    @Bind(R.id.custom_image)
    ImageView image;

    public ImageWithoutTextViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    public ImageWithoutTextViewHolder setImageResource(int resId) {
        image.setImageResource(resId);
        return this;
    }

    @Override
    public void call(Integer resId) {
       image.setImageResource(resId);
    }

    @Override
    protected String onClickCallback() {
        return null;
    }
}
